src/base.jl

```diff
@@ -200,7 +207,8 @@ function ConcreteSystem(sys::System)
         @debug "data: $(string(key)): count=$(string(length(value)))"
     end
 
-    components = _get_components_by_type(Component, data)
+    # components = _get_components_by_type(Component, data)
+    components = Dict{DataType, Any}()
```

Comment:
I think we should have a further discussions of this components construction using recursion. If we have parameterized types in PowerSystems this may pose a challenge. On a tangent we should also discuss how parametrized types should be serialized. 

I've commented this out in this PR because I didn't want to make sweeping changes across the package without having a discussion.